src/js/server/yaml.ts

```diff
@@ -0,0 +1,40 @@
+import fs from "fs";
+import yaml from "js-yaml";
+
+/**
+ * Converts a YAML string to a JSON object.
+ * @param {string} YAMLString - The YAML string to convert.
+ * @returns {object} - The resulting JSON object.
+ */
+export function convertYAMLStringToJSON(YAMLString: string): object {
```

Comment:
1. String utils should be in the `shared` folder
2. We need tests for all these